tests/unit/test_workflow_files.py

```diff
@@ -301,13 +301,32 @@ def test_infer_latest_run(
         assert infer_latest_run(path) == expected
 
 
+def test_infer_latest_run_raise_warning_for_runN(caplog, tmp_run_dir):
+    caplog.set_level(logging.WARNING, 'log')
+    cylc_run_dir: Path = tmp_run_dir()
```

Comment:
I don't think that this assignment is necessary either - `temp_run_dir` has the default fixture scope of function so you can dispense with this line if you wish, just using `tmp_run_dir()` each time you need it.